### PR TITLE
fix: guarantee test correctness during map entry iteration

### PR DESCRIPTION
**Pull request by shunfan-shao**
# Description
Test `org.apache.cayenne.util.WeakValueMapTest.testConcurrentModification` will fail under [NonDex](https://github.com/TestingResearchIllinois/NonDex) which detects flakiness under non-deterministic environment. 

The test code follows the logic below:
```java
Map<String, Integer> map = new WeakValueMap<>(3);
for(Map.Entry<String, Integer> entry : map.entrySet()) {
    if("key_2".equals(entry.getKey())) 
        map.remove("key_2");
}
```

While [WeakValueMap uses HashMap internally](https://github.com/apache/cayenne/blob/master/cayenne-server/src/main/java/org/apache/cayenne/util/ReferenceMap.java#L76), it is not guaranteed for `map.entrySet` to iterate over the entries under some orders.

The code is expected to thrown `ConcurrentModificationException` during execution. Under some edge cases, eg: `key_2` is traversed and removed last, the test will fail to throw and cause an issue.

### CODE REVIEW

File: cayenne-server/src/test/java/org/apache/cayenne/util/WeakValueMapTest.java [hunk 1]
```diff
@@ -216,9 +216,7 @@ public void testConcurrentModification() {
         assertEquals(4, map.size());
 
         for(Map.Entry<String, Integer> entry : map.entrySet()) {
```
Comment (yyfMichaelYan) [inline cayenne-server/src/test/java/org/apache/cayenne/util/WeakValueMapTest.java:218]: I think the code can be cleaner if you iterate the ```keySet``` instead of the ```entrySet```.

```
for (String key : map.keySet()) {
  map.remove(key);
}
```

Comment (shunfan-shao) [reply cayenne-server/src/test/java/org/apache/cayenne/util/WeakValueMapTest.java:218]: You are right, thanks!
